client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -222,22 +224,21 @@ const CancelPurchaseButton = React.createClass( {
 		} );
 
 		if ( config.isEnabled( 'upgrades/removal-survey' ) ) {
-			const { purchase } = this.props,
-				surveyData = {
-					'why-cancel': {
-						response: this.state.survey.questionOneRadio,
-						text: this.state.survey.questionOneText
-					},
-					'next-adventure': {
-						response: this.state.survey.questionTwoRadio,
-						text: this.state.survey.questionTwoText
-					},
-					'what-better': { text: this.state.survey.questionThreeText },
-					purchase: purchase.productSlug,
-					type: 'refund'
-				};
-
-			submitSurvey( 'calypso-remove-purchase', this.props.selectedSite.ID, surveyData );
+			const { purchase, selectedSite } = this.props;
+			const surveyData = {
+				'why-cancel': {
+					response: this.state.survey.questionOneRadio,
+					text: this.state.survey.questionOneText
+				},
+				'next-adventure': {
+					response: this.state.survey.questionTwoRadio,
+					text: this.state.survey.questionTwoText
+				},
+				'what-better': { text: this.state.survey.questionThreeText },
+				type: 'refund'
+			};
+
+			submitSurvey( 'calypso-remove-purchase', this.props.selectedSite.ID, enrichedSurveyData( surveyData, moment(), selectedSite, purchase ) );
```

Comment:
it would probably be nice to split this into multiple lines, maybe one per argument…